Configure

```diff
@@ -1281,6 +1281,26 @@ if ($^O ne "VMS") {
 	    $disabled{makedepend} = "unavailable" unless $config{makedepprog};
 	}
     }
+
+    if (!$disabled{asm}) {
+	# probe for -Wa,--noexecstack option...
+	if ($predefined{__clang__}) {
+	    # clang [which has builtin assembler] seems to recognize the
+	    # option in question on all supported platforms even when
+	    # it's meaningless...
+	    $config{cflags} .= " -Wa,--noexecstack";
+	} else {
+	    open(PIPE, "$cc -Wa,--help -c -o null.$$.o -x assembler /dev/null 2>&1 |");
+	    while(<PIPE>) {
+		if (m/\-\-noexecstack/) {
+		    $config{cflags} .= " -Wa,--noexecstack";
+		    last;
+		}
+	    }
+	    close(PIPE);
+	    unlink("null.$$.o");
+	}
+    }
```

Comment:
No error check on the open will surely break on windows and certainly on VMS.

Reply:
Also, I would prefer if Configure was as platform agnostic as possible. This is a change in the opposite direction

Reply:
> VMS

This whole thing is in `$^O ne "VMS"`

> Windows

Why would it break? cl will react "violently" on gcc options, but it works out, since it won't report --noexecstack. One can argue that check could be conditional on `$predefined{__GNUC__}`, but there are non-gcc compilers that recognize -Wa...

> No error check

Error would actually mean that you can't execute compiler. In such case you're in deeper trouble.

> as platform agnostic as possible

Life is compromise and the only question if any given one is reasonable...

Reply:
One can argue that it could be moved to Configurations, but then trouble is that it can be version-specific...

Reply:
I've some work going on that might help, hold on...

Reply:
> I've some work going on that might help

If it's #4684, then I fail to see how it's related. Well, I fail to see how it's *directly* related.

One can solve it differently by arranging subroutine in 10-main.conf:vc_win\*_info style that does the probe and refer to it from platform configs... Is this it? Just in case once again, idea behind making it a config-time *probe* is not to make assumption that all binutils versions on any given platform support it.

Reply:
Yes, that's the idea.

I'm currently working on compiler (or rather, toolchain) templates, which will show in a separate PR, which should also make things simpler in terms of duplicating stuff to a gazillion configs (for example all those that expect gcc or clang).